JMGuildSaleHistoryTracker.lua

```diff
@@ -250,6 +256,61 @@ function Indexer:addExistingDataToTheIndex()
     end
 end
 
+--[[
+
+ Sale Upgrader
+
+ Will upgrade old sales to match the newest interface
+ To fix backward compatibility of the sales
+
+ ]]
+
+local SaleUpgrader = {}
+
+---
+-- Will be called after the saved variables is loaded
+-- To check and update the sale data
+--
+function SaleUpgrader:upgrade()
+    if not SavedVariables.dataVersion then
+        SavedVariables.dataVersion = 0
+    end
+
+    while SavedVariables.dataVersion < Config.dataVersion do
+        local upgradeFunction = SaleUpgrader.upgradeFunctionVersion[SavedVariables.dataVersion]
+        upgradeFunction()
```

Comment:
Maybe the intention and effects of the function are a bit clearer that way, yes.